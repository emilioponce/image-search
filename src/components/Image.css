.image {
  background-color: rgb(238, 238, 238);
  box-shadow: 7px 7px 10px 0px rgba(0, 0, 0, 0.15);
}

.caption {
  font-size: 80%;
}
