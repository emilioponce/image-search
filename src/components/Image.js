import React, { Component } from 'react'

import './Image.css'

const Image = props => {
  return (
    <div className="Image">
      <div>
        <img src="https://via.placeholder.com/150" />
      </div>
      <div>{props.caption}</div>
    </div>
  )
}

export default Image
