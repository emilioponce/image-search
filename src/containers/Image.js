import React, { Component } from 'react'
import PropTypes from 'prop-types'
import Lightbox from 'react-image-lightbox'
import moment from 'moment'

import 'react-image-lightbox/style.css'
import './Image.css'

class Image extends Component {
  constructor(props) {
    super(props)
    this.state = {
      isOpen: false
    }
    this.openLightBox = this.openLightBox.bind(this)
    this.closeLightBox = this.closeLightBox.bind(this)
  }

  openLightBox = () => {
    this.setState({ isOpen: true })
  }

  closeLightBox = () => {
    this.setState({ isOpen: false })
  }

  render = () => {
    const { url, urlBig, title, date, owner, description } = this.props
    const { isOpen } = this.state

    const humanReadableDate = moment.unix(date).format('MM/DD/YYYY')

    return (
      <div className="image" onClick={this.openLightBox}>
        <div>
          <img src={url} />
        </div>
        <div className="caption">
          {title.length > 20 ? title.slice(0, 15).concat(' ...') : title}
        </div>
        {isOpen && (
          <Lightbox
            mainSrc={urlBig}
            onCloseRequest={this.closeLightBox}
            imageTitle={title}
            imageCaption={`${
              description._content
            } (Owner: ${owner}. Uploaded: ${humanReadableDate}) `}
          />
        )}
      </div>
    )
  }
}

Image.propTypes = {
  url: PropTypes.string,
  urlBig: PropTypes.string,
  title: PropTypes.string,
  date: PropTypes.string,
  owner: PropTypes.string,
  description: PropTypes.object
}

export default Image
