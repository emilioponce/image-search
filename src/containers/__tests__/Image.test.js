import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'

import Image from '../Image'

describe('Image component', () => {
  const props = {
    id: jest.fn(),
    url_q: 'https://farm2.staticflickr.com/1968/30605441267_65d5469660_q.jpg',
    urlBig: 'https://farm2.staticflickr.com/1968/30605441267_65d5469660.jpg',
    title: 'title text',
    dateupload: '1541339304',
    ownername: 'owner',
    decription: 'image description'
  }

  const component = (
  <Image
    key={props.id}
    url={props.url_q}
    urlBig={props.url_m}
    title={props.title}
    date={props.dateupload}
    owner={props.ownername}
    description={props.description}
  />
  )

  it('Renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(component, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  it('Snapshot matchs', () => {
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
