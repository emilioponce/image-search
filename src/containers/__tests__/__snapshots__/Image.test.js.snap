// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image component Snapshot matchs 1`] = `
<div
  className="image"
  onClick={[Function]}
>
  <div>
    <img
      src="https://farm2.staticflickr.com/1968/30605441267_65d5469660_q.jpg"
    />
  </div>
  <div
    className="caption"
  >
    title text
  </div>
</div>
`;
